### PR TITLE
Product Page: Add Discounted Product Tests

### PR DESCRIPTION
**Pull request by ardelato**
## Description

This adds Jest tests to address #1132

## CR Notes:

I opted out of writing a `Jest` test to confirm the cart price matches the listed price when adding to the cart because it will be much simpler to test this through a `Cypress`, or `Playwright`, than through a `Jest` test. 

## QA Notes:

- [ ] CI should pass

Connects: #1132

### CODE REVIEW

**Comment by k0rvusk0r4x** [comment]
- CI does, indeed, pass.
- Nothing jumps out at me as "Wait, what about . . . " from a QA perspective.

QA 😺

File: frontend/tests/jest/tests/ProductSection.test.tsx [hunk 1]
```diff
@@ -137,5 +138,49 @@ describe('ProductSection Tests', () => {
...
+            (child: HTMLElement) => {
+               return child !== price;
+            }
+         );
+
+         const priceTexts = [discountPercentage + '% OFF', '$' + originalPrice];
+
+         siblings.forEach((sibling: HTMLElement) => {
+            (expect(priceTexts) as any).toContain(sibling.textContent);
+         });
```
[the remark this answers is not in the document]
Comment (ardelato) [reply frontend/tests/jest/tests/ProductSection.test.tsx:183]: Yeah it's a bit of a hack, but this means we don't need to add additional **data-testid** attributes to the other elements

File: frontend/tests/jest/tests/ProductSection.test.tsx [hunk 2]
```diff
@@ -4,112 +4,183 @@ import {
...
-      );
-
-      const warning = screen.queryByTestId('product-warning');
-      const note = screen.queryByTestId('product-note');
-      const disclaimer = screen.queryByTestId('product-disclaimer');
-
-      (expect(warning) as any).not.toBeInTheDocument();
-      (expect(note) as any).not.toBeInTheDocument();
-      (expect(disclaimer) as any).not.toBeInTheDocument();
+      test.only('Discounted Price Renders', async () => {
```
Comment (batbattur) [inline frontend/tests/jest/tests/ProductSection.test.tsx:142]: @ardelato I just noticed this `test.only`. Is there a reason why we're doing this?

Comment (ardelato) [reply frontend/tests/jest/tests/ProductSection.test.tsx:142]: Ahh shoot! That was left over from testing only this test 😓

Comment (ardelato) [reply frontend/tests/jest/tests/ProductSection.test.tsx:142]: I'll open a follow-up pull to address this. Nice catch @batbattur

Comment (ardelato) [reply frontend/tests/jest/tests/ProductSection.test.tsx:142]: https://github.com/iFixit/react-commerce/pull/1150

**Pull request merged**